lib/types/settlements.d.ts

```diff
@@ -0,0 +1,343 @@
+         */
+        fees: number;
+        /**
+         * Total tax, in paise, charged for the fee component.
+         */
+        tax: number;
+        /**
+         * The 3-letter ISO currency code for the settlement.
+         */
+        currency: string;
+        /**
+         * Indicates the state of the instant settlement
+         */
+        status: 'created' | 'processed' | 'failed' | 'initiated' | 'reversed' | 'partially_processed';
+        /**
+         * Unix timestamp at which the instant settlement was created.
+         */
+        created_at: number;
+        /**
+         * List of payouts created for the instant settlement.
+         */
+        ondemand_payouts?: {
+            entity: string;
+            count: number;
+            items: Settlements.RazorpaySettlement[];
+        }
+        scheduled: boolean;
+    }  
+}
+
+declare function settlements(api: any): {
+    /**
+     * Create on-demand settlement
+     * 
+     * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#create-an-instant-settlement) for required params
+     * 
+     */
+    createOndemandSettlement(params: Settlements.RazorpayInstantSettlementCreateRequestBody): Promise<Settlements.RazorpayInstantSettlement>
+    createOndemandSettlement(params: Settlements.RazorpayInstantSettlementCreateRequestBody, callback: (err: INormalizeError | null, data: Settlements.RazorpayInstantSettlement) => void): void
+    /**
+    * Get all settlements
+    *
+    * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#fetch-all-instant-settlements) for required params
+    *
+    */
+    all(params?: RazorpayQuery): Promise<{
+        entity: string;
+        count: number;
+        items: Array<Settlements.RazorpaySettlement>;
+    }>
+    all(params: RazorpayQuery, callback: (err: INormalizeError | null, data: {
+        entity: string;
+        count: number;
+        items: Array<Settlements.RazorpaySettlement>;
+    }) => void): void
+    /**
+    * Fetches a settlement given Settlement ID
+    *
+    * @param settlementId - The unique identifier of the settlement.
+    *
+    */
+    fetch(settlementId: string): Promise<Settlements.RazorpaySettlement>
+    fetch(settlementId: string, callback: (err: INormalizeError | null, data: Settlements.RazorpaySettlement) => void): void
+    /**
+    * Fetch all demand settlements
+    *
+    * @param settlementId - The unique identifier of the settlement.
+    * @param params - Check [doc](https://razorpay.com/docs/api/settlements/instant#fetch-all-instant-settlements) for required params
+    * 
+    */
+    fetchAllOndemandSettlement(params: RazorpayQuery | { 'expand[]': 'ondemand_payouts' }): Promise<{
+        entity: string;
+        count: number;
+        items: Settlements.RazorpayInstantSettlement[];
+    }>
+    fetchAllOndemandSettlement(params: RazorpayQuery | { 'expand[]': 'ondemand_payouts' }, callback: (err: INormalizeError | null, data: {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Why are these mutually exclusive? Isn't expand an optional parameter?